packages/flutter_tools/lib/src/project_validator.dart

```diff
@@ -2,16 +2,129 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'base/file_system.dart';
+import 'base/logger.dart';
+import 'flutter_manifest.dart';
 import 'project.dart';
 import 'project_validator_result.dart';
 
 abstract class ProjectValidator {
   String get title;
   bool supportsProject(FlutterProject project);
   /// Can return more than one result in case a file/command have a lot of info to share to the user
-  Future<List<ProjectValidatorResult>> start(FlutterProject project);
+  Future<List<ProjectValidatorResult>> start(FlutterProject project, {required Logger logger, required FileSystem fileSystem});
   /// new ProjectValidators should be added here for the ValidateProjectCommand to run
-  static const List <ProjectValidator> allProjectValidators = <ProjectValidator>[
-    // TODO(jasguerrero): add validators
+  static List <ProjectValidator> allProjectValidators = <ProjectValidator>[
+    GeneralInfoProjectValidator(),
   ];
 }
+
+/// Validator run for all platforms that extract information from the pubspec.yaml
+///
+/// Specific info from different platforms should be written in their own ProjectValidator
+class GeneralInfoProjectValidator extends ProjectValidator{
+  @override
+  Future<List<ProjectValidatorResult>> start(FlutterProject project, {required Logger logger, required FileSystem fileSystem}) async {
+    final FlutterManifest? flutterManifest = FlutterManifest.createFromPath(
+        project.pubspecFile.path,
+        logger: logger,
+        fileSystem: fileSystem
+    );
+    if (flutterManifest == null) {
+      return <ProjectValidatorResult>[_emptyProjectValidatorResult];
+    }
+    final ProjectValidatorResult appNameValidatorResult = getAppNameResult(flutterManifest);
+    final String supportedPlatforms = getSupportedPlatforms(project);
+    if (supportedPlatforms.isEmpty) {
+      return <ProjectValidatorResult>[appNameValidatorResult];
```

Comment:
Rather than returning a list literal, can you create an empty list at the start of this function, adding to it as you go, and if we need to return early, return that reference. This will be more resilient in the future when we add more validators, we won't have to update all subsequent list literals.